lib/workqueue.c

```diff
@@ -90,6 +90,13 @@ void work_queue_free(struct work_queue *wq)
 	if (wq->thread != NULL)
 		thread_cancel(wq->thread);
 
+	while(!work_queue_empty(wq)) {
+		struct work_queue_item *item = work_queue_last_item(wq);
+
+		work_queue_item_dequeue(wq, item);
+		work_queue_item_free(item);
```

Comment:
Any reason for not using `work_queue_item_remove()` instead?

Reply:
Because I don't have a item already.  I just know I have something on it.